A51-xds-custom-backend-metrics.md

```diff
@@ -0,0 +1,382 @@
+A51: xDS Custom Backend Metrics Support
+----
+* Author(s): [Yifei Zhuang](https://github.com/YifeiZhuang), [Mark Roth](https://github.com/markdroth)
+* Approver: Eric Anderson
+* Status: Ready for Implementation
+* Implemented in: <language, ...>
+* Last updated: 2022-02
+* Discussion at: https://groups.google.com/g/grpc-io/
+
+## Abstract
+Support custom metrics injection at a gRPC server, and consumption at a gRPC client. This is useful 
+for customers to implement sophisticated client load balancing strategies. 
+For example, the client takes into account the
+backend server CPU and memory utilization metrics to do its routing. A few pieces are involved here: 
+
+* The API surface exposed to users to enable the metrics injection and consumption system.
+* The two backend report collection mechanisms and the gRPC internal plumbing using its underlying infrastructures.
+* Metrics, the wire formats, and use cases.
+
+
+## Background
+
+Previously xDS load balancer has to be configured to one of the built-in policies: e.g. round-robin or ring hash.
+The [custom load balancer proposal](link place holder) makes it possible to plug in users' own LB policy. 
+However, customers' own LB policy typically depends on extra input, e.g. CPU utilization, backend queue size.
+In fact, many fine-tuned LB policies requires feedback signals from the
+backend to the data plane clients, e.g. weighted-round-robin LB policy.
+To fill the gap and complements custom LB policy, gRPC now brings APIs to allow users to inject custom 
+metrics at the server and consume the metrics from the client.
+The [ORCA](https://github.com/envoyproxy/envoy/issues/6614) proposal fits perfectly into the solution. 
+We propose to use ORCA service and message standards defined in UDPA protos.
+
+### Related Proposals:
+
+* [A36: xDS-Enabled Servers][A36]
```

Comment:
I don't think we need to reference either gRFC A36 or the upcoming xDS custom LB policy design here.